api/collections/serializers.py

```diff
@@ -0,0 +1,113 @@
+from rest_framework import serializers as ser
+from api.base.serializers import JSONAPISerializer, CollectionLinksField, Link, LinksField
+from website.models import Node
+from framework.auth.core import Auth
+
+class CollectionSerializer(JSONAPISerializer):
+    filterable_fields = frozenset(['title'])
+
+    id = ser.CharField(read_only=True, source='_id')
+    title = ser.CharField(required=True)
+    date_created = ser.DateTimeField(read_only=True)
+    date_modified = ser.DateTimeField(read_only=True)
+    modified_by = ser.CharField(read_only=True, source='get_modified_by')
+    links = CollectionLinksField({
+        'children': {
+            'related': Link('collections:collection-children', kwargs={'pk': '<pk>'}),
+            'count': 'get_node_count',
+        },
+        'pointers': {
+            'related': Link('collections:collection-pointers', kwargs={'pk': '<pk>'}),
+            'count': 'get_pointers_count',
+        },
+    })
+    properties = ser.SerializerMethodField(help_text='A dictionary of read-only booleans: registration, collection,'
+                                                     'and dashboard. Collections are special nodes used by the Project '
+                                                     'Organizer to, as you would imagine, organize projects. '
+                                                     'A dashboard is a collection node that serves as the root of '
+                                                     'Project Organizer collections. Every user will always have '
+                                                     'one Dashboard')
+    # TODO: finish me
+
+    class Meta:
+        type_ = 'collections'
+
+    def get_node_count(self, obj):
+        auth = self.get_user_auth(self.context['request'])
+        nodes = [node for node in obj.nodes if node.can_view(auth) and node.primary]
+        return len(nodes)
+
+    def get_pointers_count(self, obj):
+        return len(obj.nodes_pointer)
+
+    def get_user_auth(self, request):
+        user = request.user
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        return auth
+
+    @staticmethod
+    def get_properties(obj):
+
+        user = obj.logs[-1].user
+        modified_by = user.family_name or user.given_name
+
+        ret = {
+            'collection': obj.is_folder,
+            'dashboard': obj.is_dashboard,
+            'smart_folder': obj.smart_folder,
+            'modified_by': modified_by,
```

Comment:
The help text on Properties indicates that these are all booleans. So we either need to move modified_by out of properties, or we need to change the help text everywhere. I think go ahead and move it out to its own field, as we may have to do other large-scale restructuring of all the data, and no sense going to crazy right now.